subsystems/networking/bluetooth_tethering_linux.go

```diff
@@ -0,0 +1,255 @@
+	return b.RequestAuthorization(devicePath)
+}
+
+// generic requests, we compare the HW address to accept/deny.
+func (b *basicAgent) RequestAuthorization(devicePath dbus.ObjectPath) *dbus.Error {
+	conn, _, err := getBluetoothDBus()
+	if err != nil {
+		b.logger.Error(err)
+		return errPairingRejected
+	}
+
+	remoteDev := conn.Object(BluezDBusService, devicePath)
+
+	ret, err := remoteDev.GetProperty("org.bluez.Device1.Address")
+	if err != nil {
+		b.logger.Error(err)
+		return errPairingRejected
+	}
+	bdaddr := strings.ToUpper(ret.Value().(string))
+
+	ret, err = remoteDev.GetProperty("org.bluez.Device1.Alias")
+	if err != nil {
+		b.logger.Error(err)
+	}
+	alias := ret.Value().(string)
+
+	b.logger.Infof("Bluetooth pairing request from: %s (%s)", bdaddr, alias)
+
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	trusted, ok := b.trusted[bdaddr]
+	if !(ok && trusted) && !b.trustAll && !b.trustAllTemp {
+		b.logger.Errorf("Bluetooth device pairing rejected for %s (%s), device address must be added via provisioning first.", bdaddr, alias)
+		return errPairingRejected
+	}
+
+	tetherCfg := utils.NetworkDefinition{
+		Type:      NetworkTypeBluetooth,
+		Interface: bdaddr,
+	}
+	_, err = b.networking.AddOrUpdateConnection(tetherCfg)
+	if err != nil {
+		b.logger.Error(errw.Wrapf(err, "adding/updating tethering config for %s (%s)", bdaddr, alias))
+	} else {
+		b.logger.Infof("Added network manager profile for tethering with %s (%s)", bdaddr, alias)
+	}
+
+	if err := remoteDev.SetProperty("org.bluez.Device1.Trusted", true); err != nil {
+		b.logger.Error(errw.Wrapf(err, "setting trust property for bluetooth device %s (%s)", bdaddr, alias))
+	}
+
+	b.workers.Add(1)
+	ctx, cancel := context.WithTimeout(context.Background(), connectTimeout)
+	b.cancel = cancel
+	go func() {
+		defer cancel()
+		defer b.workers.Done()
+
+		var paired bool
+		defer func() {
+			switch {
+			case b.networking.connState.getOnline():
+				b.logger.Infof("Bluetooth tethering fully online.")
+			case paired:
+				b.logger.Warnf("Failed to fully connect to the internet via tethering. Will keep trying. " +
+					"Please retry pairing if not online in five minutes.")
+			default:
+				b.logger.Warnf("Failed to complete bluetooth pairing. Please retry the pairing process.")
+			}
+		}()
+
+		for !paired {
+			// break if our context times out or is cancelled
+			if !b.networking.mainLoopHealth.Sleep(ctx, time.Second) {
+				b.logger.Warn("timed out waiting for bluetooth pairing to complete")
+				return
+			}
+
+			ret, err := remoteDev.GetProperty("org.bluez.Device1.Paired")
+			if err != nil {
+				b.logger.Warn(errw.Wrapf(err, "cannot get paired status for bluetooth device: %s (%s)", bdaddr, alias))
+			}
+			paired = ret.Value().(bool)
+		}
+
+		// We have to connect something to make service discovery figure out tethering, so we'll just try an invalid service
+		call := remoteDev.CallWithContext(ctx, "org.bluez.Device1.ConnectProfile", 0, "deadbeef-cafe-0000-0000-cafedeadbeef")
```

Comment:
i'm not sure i follow why is this necessary, is this so the phone can find the device?